packages/core/test/unit/query-generator/add-index-query.test.ts

```diff
@@ -0,0 +1,282 @@
+      default: 'CREATE INDEX [table_column1] ON [table] ([column1])',
+      sqlite: 'CREATE INDEX `table_column1` ON `table` (`column1` COLLATE `en_US`)',
+      postgres: 'CREATE INDEX "table_column1" ON "table" ("column1" COLLATE "en_US")',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column1`)',
+    });
+  });
+
+  it('generates a query that adds an index with column length', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: [{ name: 'column1', length: 7 }] }), {
+      default: 'CREATE INDEX [table_column1] ON [table] ([column1])',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column1`(7))',
+    });
+  });
+
+  it('generates a query that adds an index with column order', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: [{ name: 'column1', order: 'DESC' }] }), {
+      default: 'CREATE INDEX [table_column1] ON [table] ([column1] DESC)',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column1` DESC)',
+    });
+  });
+
+  it('generates a query that adds an index with column operator', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: [{ name: 'column', operator: 'inet_ops' }] }), {
+      default: 'CREATE INDEX [table_column] ON [table] ([column])',
+      postgres: 'CREATE INDEX "table_column" ON "table" ("column" inet_ops)',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column`)',
+    });
+  });
+
+  it('generates a query that adds an index with column collate, length, operator and order', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: [{ name: 'column', collate: 'en_US', length: 7, order: 'DESC', operator: 'inet_ops' }] }), {
+      default: 'CREATE INDEX [table_column] ON [table] ([column] DESC)',
+      sqlite: 'CREATE INDEX `table_column` ON `table` (`column` COLLATE `en_US` DESC)',
+      postgres: 'CREATE INDEX "table_column" ON "table" ("column" COLLATE "en_US" inet_ops DESC)',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column`(7) DESC)',
+    });
+  });
+
+  it('generates a query that adds an index with column operator for multiple fields #1', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: [{ name: 'column1', order: 'DESC', operator: 'inet_ops' }, 'column2'] }), {
+      default: 'CREATE INDEX [table_column1_column2] ON [table] ([column1] DESC, [column2])',
+      postgres: 'CREATE INDEX "table_column1_column2" ON "table" ("column1" inet_ops DESC, "column2")',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column1` DESC, `column2`)',
+    });
+  });
+
+  it('generates a query that adds an index with column operator for multiple fields #2', () => {
```

Comment:
previous test could be removed in favor of only keeping this more complete one

Reply:
Updated test now